include/linux/vfs_compat.h

```diff
@@ -74,10 +74,16 @@ truncate_setsize(struct inode *ip, loff_t new)
  * 4.12 - x.y, super_setup_bdi_name() new interface.
  */
 #if defined(HAVE_SUPER_SETUP_BDI_NAME)
+extern atomic_long_t zfs_bdi_seq;
+
 static inline int
 zpl_bdi_setup(struct super_block *sb, char *name)
 {
-	return (super_setup_bdi_name(sb, name));
+	char tmp[32];
+
+	sprintf(tmp, "%.28s%s", name, "-%d");
+	return super_setup_bdi_name(sb, tmp,
+	    atomic_long_inc_return(&zfs_bdi_seq));
```

Comment:
There's no reason to use two format functions, fold sprintf into `super_setup_bdi_name`